convertercurrency/src/main/java/page/devnet/convertercurrency/appcurrencyapi/CurrencyapiService.java

```diff
@@ -0,0 +1,133 @@
+package page.devnet.convertercurrency.appcurrencyapi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.protocol.HTTP;
+import page.devnet.convertercurrency.ConverterCurrencyException;
+import page.devnet.convertercurrency.ConverterCurrencyService;
+import page.devnet.convertercurrency.utils.ParserCurrencyMessage;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Konstantin Agafonov
+ * @since 01.08.24
+ */
+@Slf4j
+//TODO
+public class CurrencyapiService implements ConverterCurrencyService {
+
+    private final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+    private final URI appCurrencyapiUrl;
+    private final HttpClient client;
+    private final ParserCurrencyMessage parserCurrencyMessage = new ParserCurrencyMessage();
+
+    private static final List<String> currencys = Arrays.asList("RUB", "USD", "EUR", "JPY", "KZT", "GEL", "NZD");
+
+    public CurrencyapiService(String apiKey) {
+        try {
+            this.appCurrencyapiUrl = new URIBuilder(URI.create("https://api.currencyapi.com/v3/latest"))
+                    .addParameter("key", apiKey)
+                    .build();
+        } catch (URISyntaxException e) {
+            log.error(e.getMessage(), e);
+            throw new ConverterCurrencyException(e);
+        }
+
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(5000)
+                .setConnectionRequestTimeout(5000)
+                .setSocketTimeout(5000).build();
```

Comment:
i recommend exctract 5000 to system envirenment or at least in static final fields